support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -301,7 +330,7 @@ object DigitalSubscriptionGiftRedemption {
     } yield Success(newTermLength)).getOrElse(Failure(new RuntimeException(s"Unable to calculate new term length for subscription ${subscription}")))
```

Comment:
Just something to bear in mind rather than a request to change - it's great to standardise on Try/Future to reduce boilerplate in the for comprehensions, but I can't help thinking all the "new RuntimeException" is just boilerplate elsewhere.  If we were implementing this function in isolation we would just use `Either[String, Int]`

Reply:
Maybe, although it does have to be converted to an exception at some point so it might as well be here

Reply:
fair enough - this was the thing that prompted me to think about the balance keeping the modules self contained e.g. GiftCodeGenerator vs keeping them in the overall structure i.e. avoiding loads of conversions in the for comprehensions